rust/benches/l2.rs

```diff
@@ -39,10 +39,9 @@ fn l2_arrow_arity(x: &Float32Array, y: &Float32Array) -> f32 {
 }
 
 #[inline]
-fn l2_auto_vectorization(x: &Float32Array, y: &Float32Array) -> f32 {
-    x.values()
-        .iter()
-        .zip(y.values().iter())
+fn l2_auto_vectorization(x: &[f32], y: &[f32]) -> f32 {
+    x.iter()
+        .zip(y.iter())
         .map(|(a, b)| (a - b).powi(2))
```

Comment:
Is it faster to do a-b times itself instead of power?

Reply:
https://rust.godbolt.org/z/KG48nn6ec  as you can see from the compiler output, this `powi(2)` is compiled to `vmulss`

```Assembly 
vsubss  xmm1, xmm1, dword ptr [rdx + 4*rsi + 24]
vmulss  xmm1, xmm1, xmm1
vaddss  xmm0, xmm0, xmm1
```

which is the similar as `(a-b) * (a-b)` (https://rust.godbolt.org/z/f67vn6PbW)

However the former has better loop unrolling.